app/src/main/java/cm/aptoide/pt/search/SearchRepository.kt

```diff
@@ -51,59 +50,69 @@ class SearchRepository(val storeRepository: RoomStoreRepository,
   }
 
   fun generalSearch(query: String, filters: SearchFilters,
-                    matureEnabled: Boolean): Completable {
+                    matureEnabled: Boolean,
+                    useCachedValues: Boolean): Completable {
     return Single.just(query)
-        .flatMapCompletable { search(query, filters, matureEnabled, null) }
+        .flatMapCompletable { search(query, filters, matureEnabled, null, useCachedValues) }
         .subscribeOn(Schedulers.io())
   }
 
-  fun searchInStore(query: String, filters: SearchFilters, matureEnabled: Boolean,
-                    storeName: String): Completable {
+  fun searchInStore(query: String, filters: SearchFilters,
+                    matureEnabled: Boolean,
+                    storeName: String,
+                    useCachedValues: Boolean): Completable {
     return Single.just(query)
-        .flatMapCompletable { search(query, filters, matureEnabled, storeName) }
+        .flatMapCompletable { search(query, filters, matureEnabled, storeName, useCachedValues) }
         .subscribeOn(Schedulers.io())
   }
 
   @Synchronized
   private fun search(query: String, filters: SearchFilters,
-                     matureEnabled: Boolean, specificStore: String?): Completable {
-    cachedSearchResults?.let { activeResults ->
-      if (activeResults.query == query && activeResults.specificStore == specificStore
-          && filters == activeResults.filters && !activeResults.hasError()) {
-        if (activeResults.hasMore()) {
-          if (loadingMore) {
-            return Completable.complete()
+                     matureEnabled: Boolean, specificStore: String?,
+                     useCachedValues: Boolean): Completable {
+    if (useCachedValues) {
+      return Completable.fromAction {
+        resultsSubject.onNext(cachedSearchResults)
+      }
+    } else {
```

Comment:
Personally, and you don't have to change if you don't want to, I would remove the 'else'. Simply because indenting big blocks of code doesn't really help readibility down the line, especially if there are inner if blocks